pkg/steps/multi_stage.go

```diff
@@ -352,7 +355,15 @@ func (s *multiStageTestStep) runSteps(
 const multiStageTestStepContainerName = "test"
 
 func (s *multiStageTestStep) generatePods(steps []api.LiteralTestStep, env []coreapi.EnvVar,
-	hasPrevErrs bool) ([]coreapi.Pod, error) {
+	hasPrevErrs bool) ([]coreapi.Pod, func(string) bool, error) {
+	bestEffort := sets.NewString()
+	isBestEffort := func(podName string) bool {
+		if s.allowBestEffortPostSteps == nil || (s.allowBestEffortPostSteps != nil && !*s.allowBestEffortPostSteps) {
```

Comment:
Why not just return an empty set from the outer function when this condition is true?

Reply:
```suggestion
		if s.allowBestEffortPostSteps == nil || !*s.allowBestEffortPostSteps {
```